docs/proposals/kep-003-service-resouce-aggregation.md

```diff
@@ -0,0 +1,57 @@
+
+Furthermore, catapult component handles the webhooks for internal apiGroup. If there is `CustromResourceDiscoverySet` there's a new catapult deployment per each service cluster offering the same service.
+
+* See also:
+
+https://github.com/kubermatic/kubecarrier/issues/316
+
+## Issues
+
+The issues with this approach are multi-fold:
+
+* propagation is asynchronously, whereas management clusters serves as a "cache" to service cluster's ground truth. In more complicated setups the management cluster's resource version and service cluster resource version could diverge in unreconciable way. To give a simple, albeit toy example consider the following:
+
+    * There's deployment on the service cluster. This resource is propagated to the management cluster. At the start the deployment replicas is 1.
+       * HPA controller on the service cluster increases the replica count to 2
+       * User on the management cluster changes the deployment image
+
+   Now we have to two diverging versions what the deployment should be, and how should it look like. Somehow we have to perform 2-way, or 3-way merge between those objects adding additional complexity.
+
+   Note: In the common case the service cluster operators would only change the `.status` subresource, while the management cluster would handle the `.spec` field.
+
+* webhook complexity
+
+    We're running semi-complex code for object updating and creation webhooks. We rely on `dry-run` support from the service cluster. Service cluster resource might not support some critical `dry-run` operation, thus our webhooks won't behave as if they are being run in the service cluster in the first place.
+
+* sysadmin operator automation
+    * Where `CustomResourceDiscoverySet` is used for exposing same service from multiple service cluster, there's no unified resource on the management cluster for automation. There's `db.internal.svc-1.provider-1`, `db.internal.svc-2.provider-1`, ... api groups. The sysadmin wants something simple, `db.internal.provider-1` apiGroup smartly aggregating each service cluster resource into an unified apiGroup.
+
+* resource consumption
+    * multiple deployments per `CustomResourceDiscovery` start stacking up, increasing the KubeCarrier solution heaviness and limiting efficient resource allocation. Lean is green.
+
+
+## Proposed solution
+
+Instead of creating CRDs for each `CustomResourceDiscovery`, and deploying catapult per `CustomResourceDiscovery`, we deploy a single `neocatapult` per provider. This `neocatapult` is API Exstension server smartly proxying each API group to the service cluster, as well aggregating multiple service cluster under same apiGroup. The benefits are as follows:
```

Comment:
How do we dynamically register new apis to the `neocatapult` for provider?

Reply:
`APIServices` resource? 

This is unknown unknown (read I haven't tried it), but I think I can add additional APIs during runtime to the API extension server